fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexDeferredMaintenancePolicy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * IndexDeferredMaintenancePolicy.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+/**
+ * Some store's indexes may need merging on some occasions. This helper module should allow the caller
+ * to set and probe the merge policy and merge requests.
+ */
+public class IndexDeferredMaintenancePolicy {
+    private boolean mergeRequired = false;
+    private boolean autoMergeDuringCommit = true;
```

Comment:
I am skeptical that treating all index merges as equivalent is going to work out. I know it's hard to say when there is only one example. But it would feel more natural to me to have per-index-type settings. Or even per index if a caller chooses. Either with subclassing of the policy or maps.

Reply:
At least one other potential use case would be an extension to allow `deleteWhere` on some value indexes that wouldn't otherwise support the query, but instead to the delete asynchronously, and handle the inconsistency under the hood. That does feel as though it has some difference, in that it *can't* be done in the transaction, whereas the lucene merges can.
That being said, it seems like this is trying to accomplish two things:
1. Declaring to the record layer that the caller will take care of this asynchronous work, and provide a mechanism for how to do so
2. A policy as to what should be done asynchronously